### DISCUSSION TITLE
List all files from given storage class

### DISCUSSION THREAD

**Discussion by struthio** [moosefs/moosefs#451] [Q&A]
hi,

Is there a way of listing all files that have some storage class assigned ?
Right now I can check single file:
```
> mfsgetsclass ./file.1
./file.1: 'sclass1'
```

I can go with whole directory:
```
> mfsgetsclass ./*
./file.1: 'sclass1'
./file.2: 'sclass2'
...
```

but when I run it on fill root, i get just summary
```
> mfsgetsclass -r .
files with storage class       'sclass1' :      99405
files with storage class       'sclass2' :       2846
(...)
```
but how to get ex. all files that have sclass2 assigned ?

**Comment by chogata** [comment]
There is no special way to do it in MooseFS right now, you would have to do via a shell script. A simple loop in bash will get it done:

`for i in `find .`; do mfsgetsclass $i | grep "'sclass2'"; done`

(This will work as long as you don't have any files nor directories with 'sclass2' in their name, including the single quote signs. Probably not, but if you do, you might have to get a bit more creative).
Keep in mind that if you have very long directories (aka with a lot of entries in them) on your MooseFS instance, this will be a bit heavy on your master.
